### PR TITLE
Refactors artworks for-sale filter

### PR DESCRIPTION
**Pull request by gnilekaw**
Previously, the `for_sale` artworks filter showed either all artworks or only ones that were for sale. 

This PR refactors it to filter:
- all artworks when `for_sale` is null,
- only for-sale artworks when it's true,
- and only not-for-sale artworks when false

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please assign someone to your PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: components/filter2/dropdown_group/view.coffee [hunk 1]
```diff
@@ -35,7 +35,7 @@ module.exports = class DropdownGroupView extends Backbone.View
 
   toggleForSale: ->
     if @params.get('for_sale') is 'true'
-      @params.set 'for_sale', 'false'
+      @params.set 'for_sale', null
```
Comment (gnilekaw) [inline components/filter2/dropdown_group/view.coffee]: One effect of this is that it sets the query string in the URLs (on browser and to API) to `...&for_sale=&...`. I haven't seen any side effects of this, but it clutters the URLs with a blank parameter. Trimming out empty parameters from the query strings will be a deeper change though — and probably more appropriate for its own PR.

File: apps/partner/client/filter_settings.coffee [hunk 2]
```diff
@@ -3,19 +3,19 @@ module.exports =
     # A gallery's 'WORKS' section, if enabled. Includes for-sale & not-for-sale, all search facets
     'works':
       'aggregations': ['dimension_range', 'medium', 'price_range', 'total', 'for_sale']
-      'forSaleOnly': false
+      'forSale': null
       'hideForSaleButton': false
 
     # An institution's 'WORKS' section. Includes for-sale and not-for-sale works, excludes filtering based on price and saleability.
```
Comment (mzikherman) [inline apps/partner/client/filter_settings.coffee]: I think this comment needs an update- it now only includes not-for-sale works.

File: apps/partner2/client/filter_settings.coffee [hunk 3]
```diff
@@ -3,19 +3,19 @@ module.exports =
     # A gallery's 'WORKS' section, if enabled. Includes for-sale & not-for-sale, all search facets
     'works':
       'aggregations': ['dimension_range', 'medium', 'price_range', 'total', 'for_sale']
-      'forSaleOnly': false
+      'forSale': null
       'hideForSaleButton': false
 
     # An institution's 'WORKS' section. Includes for-sale and not-for-sale works, excludes filtering based on price and saleability.
```
Comment (mzikherman) [inline apps/partner2/client/filter_settings.coffee]: same, update comment pls

File: components/filter2/dropdown_group/view.coffee (hunk 1 above)
Comment (mzikherman) [inline components/filter2/dropdown_group/view.coffee]: I wonder if `@params.unset 'for_sale'` might work? I think that will remove it from the attribute hash entirely and so might clean up the URL.

**Comment by mzikherman** [comment]
Lol I still don't understand GH 'reviews'. Some of the comments got duped!

File: components/filter2/dropdown_group/view.coffee (hunk 1 above)
Comment (gnilekaw) [inline components/filter2/dropdown_group/view.coffee]: Yep, tried this. It works for the url on the API, but then the client API isn't updated. So when toggling back and forth the client stays at `?for_sale=true`.

Comment (gnilekaw) [inline components/filter2/dropdown_group/view.coffee]: Lmk if you want to explore this though. There might have been something I missed.

**Pull request merged**